### PR TITLE
release-notes.txt: add 2020.10 release notes

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This are the release notes for the upcoming 2020.10 release. Please read and ensure that your of the past 3 months work is mentioned.

### Testing procedure

Read, read, read…

### Issues/PRs references

None

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,548 @@
...
+* link object files instead of archives
+* tapsetup:
+
+    - Add support for listing bridges
+
+    - Add support to configure addresses for bridge
+
+    - Add support for configuring routes and forwarding capabilities for bridge
+
+    - Add TUN support
```
Comment (miri64) [inline release-notes.txt:210]: Something happened here when moving from the etherpad :-/

Comment (bergzand) [reply release-notes.txt]: Thanks! Amended with less newlines

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,543 @@
...
+    + generate vectors C files and IRQ defines from CMSIS headers
+    + stm32cmsis: retrieve STM32 CMSIS header from a package
+    + g0: add basic support
+    + implement periph_timer_periodic
+    * Add support for arbitrary SPI clock rate
+    * f3: fix clock configuration for HSI
+    * l0: rework clock initialization and configuration
+    * l1: rework clock initialization and configuration
+    * l4: rework clock configuration and initialization
+    * wb: rework clock configuration and initialization
```
Comment (aabadie) [inline release-notes.txt:162]:
```suggestion
    * f0/f1/f3/l0/l1/l4/wb: rework clock initialization and configurationn
```

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,543 @@
...
++ nrf52: Enable PWM on the LED pins
++ nucleo-g071rb: add initial support
++ nucleo-g070rb: add board
++ nucleo-g431rb: add initial support
++ Support for YARM board
++ add Adafruit Itsy Bitsy
+* samr21-xpro: fix reading EUI-64 from EDBG on cold boot
+* nucleo-l452re: add i2c config
+* hifive1b: add support for openocd programmer
+- stm32f4: remove default clock configuration for 96MHz and 168MHz CPUs
```
Comment (aabadie) [inline release-notes.txt:131]: I don't think this is part of the release

File: release-notes.txt [hunk 4]
```diff
@@ -1,3 +1,543 @@
...
+    * fix check for bitbanding
+    * Flush pipeline after triggering PendSV
++ cc2538: add Watchdog implementation
+* fe310:
+    + Add custom PLIC driver as peripheral
+    * Merge and optimize trap_entry with irq_arch
+    * Unify linker scripts
+* msp430: Update to inline-able IRQ API Area
+* nrf5x_common: UART: add support for 1 MBaud
+* stm32l0l1: rework clock initialization and configuration
```
Comment (aabadie) [inline release-notes.txt:151]: already mentioned in stm32 below

File: release-notes.txt [hunk 5]
```diff
@@ -1,3 +1,543 @@
...
+#12651: Failing tests on AVR (tested with atmega256rfr2-xpro)
+#12168: pkg/libb2: blake2s doesn't work on AVR
+#11885: arm7: printf() with float/double not working
+#10258: Incorrect defaultPORT building for esp32-wroom-32 on macOS
+#10122: Indeterministic hard fault in _mutex_lock(), with nRF52 SoftDevice
+#10076: cpu/cortexm_common: irq_enable returns the current state of interrupts
+        (not previous)
+#6567: periph/spi: Switching between CPOL=0,1 problems on Kinetis with software
+       CS
+#4954: chronos: compiling with -O0 breaks
```
Comment (aabadie) [inline release-notes.txt:377]: I think this issue no longer applies (see https://github.com/RIOT-OS/RIOT/issues/4954#issuecomment-718944290)

File: release-notes.txt [hunk 6]
```diff
@@ -1,3 +1,543 @@
...
+#9742: `buildtest` uses wrong build directory
+#9645: Different build behavior between `murdock` and `riot/riotbuild:latest`
+       image
+#8913: make: use of immediate value of variables before they have their final
+       value
+#8122: doxygen: riot.css modified by 'make doc'
+#6120: Windows AVR Mega development makefile Error
+#4053: macros: RIOT_FILE_RELATIVE printing wrong file name for headers
+
+Other issues (41)
```
Comment (aabadie) [inline release-notes.txt:399]: Some issues in this category could be moved to some of the categories above. They are just missing the right labels.
See [this line](https://github.com/RIOT-OS/RIOT/pull/15340/files#diff-f41b6887f65b706736aec3e6716bce4844a2aa169cba5e675c630f10b4fdbcecR416) which could be moved to `Other platforms related issues`

File: release-notes.txt [hunk 7]
```diff
@@ -1,3 +1,543 @@
...
+---
+* cortexm:
+    + Make no_thread_idle compatible with cortex-m0
+    + Add thread info to hard fault handler
+    + advertise puf_sram feature
+    * Skip context save/restore on scheduling the previous thread
+    * use mpu stack guard if DEVELHELP is enabled
+    * fix check for bitbanding
+    * Flush pipeline after triggering PendSV
++ cc2538: add Watchdog implementation
```
Comment (jeandudey) [inline release-notes.txt:144]:
```suggestion
+ cc2538: add Watchdog implementation
* cpu/cc26xx_cc13xx:
    + add & link CCFG configuration
    * enable periph clocks on sleep
    * fix typo on XOSC selection
```

File: release-notes.txt (hunk 6 above)
Comment (bergzand) [reply release-notes.txt:399]: Blame the script :)

Comment (aabadie) [reply release-notes.txt:399]: No I blame the maintainers who don't set the labels ;)

Comment (bergzand) [reply release-notes.txt:399]: > No I blame the maintainers who don't set the labels ;)

Please don't, most, if not all, of the issues are labelled. [The script is just a tad strict in which issues](https://github.com/RIOT-OS/RIOT-release-manager/blob/master/riot_release_manager.py#L557-L595) are sorted where.

File: release-notes.txt (hunk 2 above)
Comment (bergzand) [reply release-notes.txt]: Merged!

File: release-notes.txt (hunk 3 above)
Comment (bergzand) [reply release-notes.txt]: You're right, removed

File: release-notes.txt (hunk 4 above)
Comment (bergzand) [reply release-notes.txt]: Removed!

File: release-notes.txt (hunk 6 above)
Comment (bergzand) [reply release-notes.txt:395]: Maybe we should blame the author of the script :stuck_out_tongue_winking_eye:

File: release-notes.txt (hunk 5 above)
Comment (bergzand) [reply release-notes.txt]: Removed the issue and adjusted the counters

File: release-notes.txt (hunk 7 above)
Comment (bergzand) [reply release-notes.txt:143]: Added. I've left the typo fix out, is that an important enough typo to mention it in the release notes?

Comment (jeandudey) [reply release-notes.txt:143]: It isn't important enough (only a `!` typo) so let's not add it :-)

File: release-notes.txt [hunk 8]
```diff
@@ -1,3 +1,540 @@
...
+ - Support for PicoLIBC as standard C library implementation
+ - A new radio abstraction layer for ieee802.15.4 radios
+ - Improvement to the linking of modules
+ - An improved algorithm for generating local unique identifiers
+ - Pagewise addressing support for MTD devices
+ - 23 additional modules supported by Kconfig
+ - Initial rework of the clock modelling on stm32 devices
+ - 5 new boards, 2 new device drivers, 7 packages updated
+
+48 pull requests, composed of 1355 commits, have been merged since the
```
Comment (aabadie) [inline release-notes.txt:32]: only 48 PRs, are you sure ? With 1355 commits that a mean of ~28 commits per PR. There's something wrong here :)

Comment (miri64) [reply release-notes.txt:32]: Its 482 on my count, so @bergzand might just have forgotten a digit ;-)

- [465 merged to `master` between `2020.10-devel` and `2021.01-devel`](https://github.com/RIOT-OS/RIOT/pulls?q=is%3Apr+merged%3A2020-07-10T15%3A35%3A10..2020-10-04T13%3A04%3A31+base%3Amaster+is%3Amerged)
- [17 merged to `2020.10-branch`](https://github.com/RIOT-OS/RIOT/pulls?q=is%3Apr++base%3A2020.10-branch+is%3Amerged)

Comment (bergzand) [reply release-notes.txt:32]: Yup, it's wrong, it should be 484 PRs.

Comment (bergzand) [reply release-notes.txt:32]: > Its 482 on my count, so @bergzand might just have forgotten a digit ;-)

I count ~~484~~ 485 merge commits between 2020.07.1 and 2020.10-RC4. Maybe your number is technically more correct though :)

Comment (bergzand) [reply release-notes.txt]: Anyway, fixed in the latest push!

Comment (miri64) [reply release-notes.txt]: Between 2020.07.1? That includes the backports to `2020.07` I believe... though that 3 seems small.

Comment (miri64) [reply release-notes.txt]: I also get 485 when checking in the terminal...

```console
$ git log --merges --oneline 2020.10-devel..2020.10-RC4 | wc -l
485
```

Don't want to check now why ^^

**Review by aabadie — APPROVED** [review]
ACK, please squash!

**Comment by bergzand** [comment]
> ACK, please squash!

Squashed!

**Comment by miri64** [comment]
changelog.html in Circle CI [is already there](https://4612-7370241-gh.circle-artifacts.com/0/doc/changelog.html), so no need to wait for the non-required checks. The weird diff stuff on top is also there in current master (see https://github.com/RIOT-OS/RIOT/pull/15322#issuecomment-718115153).

**Comment by bergzand** [comment]
Thanks!

**Pull request merged**